lib/engine/game/g_1849/game.rb

```diff
@@ -517,13 +517,13 @@ def operating_round(round_num)
             G1849::Step::Token,
             Engine::Step::Route,
             G1849::Step::Dividend,
-            acquiring_station_tokens? ? G1849::Step::BuyToken : nil,
-            electric_dreams? ? G1849::Step::BuyEToken : nil,
+            G1849::Step::BuyToken,
+            G1849::Step::BuyEToken,
             Engine::Step::DiscardTrain,
             G1849::Step::BuyTrain,
             G1849::Step::IssueShares,
-            bonds? ? G1849::Step::BondInterestPayment : nil,
-            bonds? ? G1849::Step::Bond : nil,
+            G1849::Step::BondInterestPayment,
+            G1849::Step::Bond,
             [Engine::Step::BuyCompany, { blocks: true }],
         ].compact, round_num: round_num)
```

Comment:
You don´t need to `compact` the array any more, as it will not contain any nils.
```suggestion
        ], round_num: round_num)
```